api/src/migrations/remove-user-language.js

```diff
@@ -0,0 +1,36 @@
+const db = require('../db');
+const BATCH_SIZE = 100;
+
+const removeLanguage = (docs) => {
+  docs.forEach(doc => delete doc.language);
+  return db.medic.bulkDocs(docs);
+};
+
+const needsUpdate = (row) => Object.prototype.hasOwnProperty.call(row.doc, 'language');
+
+const fixUsers = function(skipSize) {
+  const options = {
+    include_docs: true,
+    limit: BATCH_SIZE,
+    skip: skipSize,
+    key: [ 'user-settings' ]
+  };
+  return db.medic.query('medic-client/doc_by_type', options)
+    .then(result => {
+      if (!result.rows || !result.rows.length) {
+        return Promise.resolve();
+      }
+      const rowsWithLanguage = result.rows.filter(needsUpdate).map(row => row.doc);
+      if (!rowsWithLanguage.length) {
+        return fixUsers(skipSize + BATCH_SIZE);
+      }
+      return removeLanguage(rowsWithLanguage)
+        .then(() => fixUsers(skipSize + BATCH_SIZE));
+    });
+};
+
+module.exports = {
+  name: 'remove-user-language',
```

Comment:
Thanks a lot for the migration and the integration test. 
We should also add unit tests to cover every edge case and check that batching works. Please check the other migrations unit tests here: https://github.com/medic/cht-core/tree/master/api/tests/mocha/migrations

Reply:
Hmm, maybe I am missing something, but I did add some unit tests here: https://github.com/medic/cht-core/pull/7081/files#diff-7e520661b99ef648535ac94ed1cb78f3393f1d9906ae0968695c91c3b769c638

I have gone over them again and tweaked them up a bit, but I have not been able to really come up with any more test cases that seem useful.  Let me know if I have the wrong approach here or if there are other test cases that we need!

Reply:
The one test you added is an integration test. It doesn't cover batching, it doesn't cover error handling. We should at least test that batching works correctly and that we handle errors (at least that we don't continue to query if an error happens). 

Please have a look at the folder I linked you in my comment to see how some other migrations are unit tested (for example: https://github.com/medic/cht-core/blob/master/api/tests/mocha/migrations/convert-translation-messages-fix.spec.js) 

Thanks!

Reply:
Whoops!  In my comment above I linked back to the wrong file (the integration tests).  I mean to put this link: https://github.com/medic/cht-core/pull/7081/files#diff-f9e2a40a58146b6c05a82dddd7209eaa1b0ed016e29cd1dbaa6341a731e404e6

(Or here is a non-PR link: https://github.com/jkuester/cht-core/blob/master/api/tests/mocha/migrations/remove-user-language.spec.js)  I think I have it in the correct folder `api/tests/mocha/migrations` and I tried to model it after the other tests that were there.

Reply:
Oh, this is awesome, thanks!